kwanghwi/myapp/app.js

```diff
@@ -37,20 +39,55 @@ app.get("/ping", (req, res) => {
 });
 
 app.post("/user/signup", async (req, res) => {
-  const { userId, password, email, profileImage } = req.body;
+  const { userId, name, password, email, profileImage } = req.body;
+  const saltRounds = 12;
+
+  const makeHash = async (password, saltRounds) => {
+    return await bcrypt.hash(password, saltRounds);
+  };
+
+  const hashedPassword = await makeHash(password, saltRounds);
+
   await appDataSource.query(
     `INSERT INTO users(
       user_id,
+      name,
       password,
       email,
-      profileImage
-    ) VALUES (?, ?, ?, ?);
+      profile_image
+    ) VALUES (?, ?, ?, ?, ?);
     `,
-    [userId, password, email, profileImage]
+    [userId, name, hashedPassword, email, profileImage]
   );
   return res.status(201).json({ message: "signup success!" });
 });
 
+app.post("/user/signin", async (req, res) => {
+  const { userId, password } = req.body;
+
+  const [selectQuery] = await appDataSource.query(
```

Comment:
```suggestion
  const [user] = await appDataSource.query(
```
변수명은 가급적 가장 명확하게 지어주는 것이 좋습니다. selectQuery라고 지어줄 수도 있겠지만 이는 .query() 메소드에 담긴 코드일 뿐이지 실제 이를 통해 받아 오는 data는 user object 1개 이기 때문에, `user`라고 지어주는 것이 가장 명확합니다 :)